hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/health/GatewayBaselineDao.java

```diff
@@ -57,47 +55,53 @@
-    Map<String, Baseline> baselineRequestMap = getBaselineRequestMap(request);
+    Map<String, Boolean> baselineRequestMap = getBaselineRequestMap(request);
     BaselineEntitiesRequest baselineEntitiesRequest =
         BaselineEntitiesRequest.newBuilder()
             .setStartTimeMillis(entitiesRequest.getStartTimeMillis())
             .setEndTimeMillis(entitiesRequest.getEndTimeMillis())
             .addAllEntityIds(
                 entitiesResponse.getEntityList().stream()
-                    .map(x -> x.getId())
+                    .map(Entity::getId)
                     .collect(Collectors.toList()))
             .setEntityType(entitiesRequest.getEntityType())
             .addAllBaselineAggregateRequest(
-                getFunctionExpression(entitiesRequest, baselineRequestMap))
+                getFunctionExpressionList(entitiesRequest, baselineRequestMap))
             .addAllBaselineMetricSeriesRequest(
                 getBaselineTimeSeriesRequest(entitiesRequest, baselineRequestMap))
             .build();
     return Single.just(baselineEntitiesRequest);
   }
 
-  private Map<String, Baseline> getBaselineRequestMap(EntityRequest request) {
-    Map<String, Baseline> baselineMap = new HashMap<>();
-    List<MetricRequest> metricRequests = request.metricRequests();
-    if (metricRequests.size() > 0) {
-      for (MetricRequest metricRequest : metricRequests) {
-        baselineMap.putAll(
-            metricRequest.aggregationRequests().stream()
-                .filter(x -> x.baseline() != null)
-                .collect(Collectors.toMap(x -> x.alias(), y -> y.baseline())));
-        baselineMap.putAll(
-            metricRequest.seriesRequests().stream()
-                .filter(x -> x.aggregationRequest().baseline() != null)
-                .collect(Collectors.toMap(x -> x.alias(), y -> y.aggregationRequest().baseline())));
-      }
-    }
+  private Map<String, Boolean> getBaselineRequestMap(EntityRequest request) {
+    Map<String, Boolean> baselineMap =
+        request.metricRequests().stream()
+            .flatMap(
+                metricRequest ->
+                    metricRequest.aggregationRequests().stream()
+                        .filter(MetricAggregationRequest::baseline))
+            .collect(
+                Collectors.toMap(
+                    MetricAggregationRequest::alias, MetricAggregationRequest::baseline));
+    baselineMap.putAll(
+        request.metricRequests().stream()
+            .flatMap(
+                metricRequest ->
+                    metricRequest.seriesRequests().stream()
+                        .filter(
+                            timeSeriesRequest -> timeSeriesRequest.aggregationRequest().baseline()))
+            .collect(
+                Collectors.toMap(
+                    MetricSeriesRequest::alias,
+                    seriesRequest -> seriesRequest.aggregationRequest().baseline())));
```

Comment:
IMO, avoiding nesting makes this easier to follow (also switched to set based on other comment)

```suggestion
          Stream<String> aggregationAliases = request.metricRequests().stream()
            .map(MetricRequest::aggregationRequests)
            .flatMap(Collections::stream)
            .filter(MetricAggregationRequest::baseline)
            .map(MetricAggregationRequest::alias);
            
          Stream<String> seriesAliases = request.metricRequests().stream()
            .map(MetricRequest::seriesRequests)
            .flatMap(Collections::stream)
            .filter(seriesRequests -> seriesRequests.aggregationRequest().baseline())
            .map(MetricAggregationRequest::alias);
            
          return Streams.concat(aggregationAliases, seriesAliases)
            .collect(Collectors.toUnmodifiableSet())
```

Reply:
Thanks will change accordingly